### DISCUSSION TITLE
How to remove an existing asset?

### DISCUSSION THREAD

**Discussion by MikhailGordienk** [eclipse-edc/Connector#387] [Q&A]
How one is supposed to delete an existing asset? There is `AssetLoader` for creating assets? but there is no abstraction that defines a delete method. Is it supposed to be implementation- or storage-specific and therefore implemented as extension if needed?

**Comment by DominikPinsel** [comment]
I would suggest refactoring the `DataLoader` interface to something, that allows simple CRUD operations. This would also allow to update data. Additionally I think this is not only important for `Assets`, but for other kinds of data, too.

<sub>Dominik Pinsel <dominik.pinsel@daimler.com>, Daimler TSS GmbH, [legal info/Impressum](https://github.com/Daimler/daimler-foss/blob/master/LEGAL_IMPRINT.md) </sub>

**Reply by jimmarino** [reply]
@MikhailGordienk this brings up an important topic. 

The issue is a lot more involved than creating a CRUD interface. For example, what does it mean to delete an asset that is referenced by a ContractOffer or, more importantly, a ContractAgreement? 

This should be part of a broader discussion around Asset versioning and decommissioning and I don't think it should be mixed with the discussion of more ephemeral data stored in the system.
